org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -1154,34 +1286,6 @@ class FoldingRangeCapabilities extends DynamicRegistrationCapabilities {
 	Boolean lineFoldingOnly
 }
 
-/**
- * Capabilities specific to {@code textDocument/semanticHighlighting}.
- *
- * <p>
- * <b>Note:</b> the <a href=
- * "https://github.com/Microsoft/vscode-languageserver-node/pull/367">{@code textDocument/semanticHighlighting}
- * language feature</a> is not yet part of the official LSP specification.
- *
- * @deprecated Use {@code SemanticTokens} API instead.
- */
-@Beta
-@Deprecated
-@JsonRpcData
-class SemanticHighlightingCapabilities {
```

Comment:
In theory the removal would require a major version bump on LSP4J.
Not sure about the policy here.

Reply:
I asked in #457 and one of the other admins said to remove them after the 0.10.0 release.

Reply:
It's just a question of versioning the lsp4j plugin. Does this removal lead to a major version increment or not.

Reply:
I don't deal with versioning or release management for LSP4J, but my understanding is that anything with the `@Beta` annotation can be changed (or removed) at any time. Breaking compatibility between minor versions seems to be acceptable in this project based on previous releases. I don't know if the versioning scheme used for this project is clearly defined anywhere, but I know one of the most popular versioning schemes SemVer says

https://semver.org/#spec-item-4
> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.
<br>
If anyone has strong feelings about it staying a bit longer, I would be fine with that if adoption of semantic tokens is not widespread enough.

Reply:
I've decided to create a separate PR for the removal of semantic highlighting.